src/tiled/objectselectiontool.cpp

```diff
@@ -1431,20 +1489,43 @@ void ObjectSelectionTool::saveSelectionState()
 void ObjectSelectionTool::refreshCursor()
 {
     Qt::CursorShape cursorShape = Qt::ArrowCursor;
-
+    QHash<MapObject*, MapObjectHighlight*> highlightItems;
```

Comment:
> How will this exactly solve dereferencing a roaming pointer? Or you mean to say since the second issue will be solved, this issue can not be reproduced?

Communicating the hovered object through the MapDocument is only a design change that makes the implementation easier (by moving the display to `ObjectSelectionItem`, where it is more easily kept up to date) and it would enable possible future uses like highlighting the hovered object in the Objects view at the same time. It has nothing to do with solving the roaming pointer issue.